### PR TITLE
Spring 6 and 5 compatibility

### LINKED ISSUE #75: Add support for Spring6/SpringBoot3
Spring Boot 2.7 will reach [end-of-life](https://endoflife.date/spring-boot) on 18 Nov 2023, so users of baigan library started migration to SpringBoot3/Spring6. We need to add support for SpringBoot3/Spring6.

Additionally requires migration to Java17

Spring6 migration [guide](https://github.com/spring-projects/spring-integration/wiki/Spring-Integration-5.x-to-6.0-Migration-Guide)

### PR DESCRIPTION
**Pull request by lukasniemeier-zalando**
Fixes #75 

* upgrade base Spring dependency to Spring 6
* backwards compatibility with Spring 5 (via building against Java 11)
* run build and tests against both Spring 6 and 5

We also tested the compatibility successfully with two little example applications.

### CODE REVIEW

**Comment by rameshmalla** [comment]
👍

File: README.md [hunk 1]
```diff
@@ -14,8 +14,8 @@ What makes Baigan a rockstar configuration framework ?
 	* AWS S3
 
 ## Prerequisites
-- Java 11
-- Spring framework
+- Java 17
```
Comment (muhammad-hamed) [inline README.md:17]: My assumption is that this will work also with Java 17+
```suggestion
- Java 17+
```

**Comment by muhammad-hamed** [comment]
👍

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**